src/mount.c

```diff
@@ -69,11 +69,42 @@ pu_umount(const gchar *mount_point,
     return TRUE;
 }
 
+gboolean
+pu_umount_all(const gchar *device,
+              GError **error)
+{
+    g_autofree gchar *cmd_mount = g_strdup("mount");
+    g_autofree gchar *output = NULL;
+    g_autofree gchar *expr = g_strdup_printf("^%sp[1-9]+", device);
+    g_autoptr(GRegex) regex = NULL;
+    g_autoptr(GMatchInfo) match_info = NULL;
+    gint wait_status;
+
+    g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+    if (!g_spawn_command_line_sync(cmd_mount, &output, NULL, &wait_status, error))
+        return FALSE;
+    if (!g_spawn_check_exit_status(wait_status, error))
+        return FALSE;
+
+    regex = g_regex_new(expr, G_REGEX_MULTILINE, 0, NULL);
+    g_regex_match(regex, output, 0, &match_info);
+
+    while (g_match_info_matches(match_info)) {
+        g_autofree gchar *mount = g_match_info_fetch(match_info, 0);
+        if (!pu_umount(mount, error))
+            return FALSE;
```

Comment:
Delete the mount point here

Reply:
How about we remove all directories being left after writing, *regardless* if any errors occurred or not? What do you think?

Reply:
That works as well. We could simply remove `/mnt/partup`.

Reply:
I thought about using `/run` instead of `/mnt` for mounting partitions and storing temporary data of partup. Then we would not have to worry about leaving the directory `/run/partup` there, as it is removed upon power-down (because tmpfs) and otherwise clearly belongs to partup for the lifetime of the system.

See also https://lwn.net/Articles/436012/ for more information on the `/run` directory. Maybe this fits our purposes better than `/mnt`?

Reply:
Yes, we can use `/run` as mount location instead of `/mnt`. Alternatively the mount location could be user configurable but that may be to much.